Sources/PaymentsCore/Networking/APIClient.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+public class APIClient {
+    public typealias CorrelationId = String
+
+    public var urlSession: URLSession
+    public var environment: Environment
+
+    public init(urlSession: URLSession = .shared, environment: Environment) {
+        self.urlSession = urlSession
+        self.environment = environment
+    }
+
+    public func fetch<T: APIRequest>(
+        endpoint: T,
+        completion: @escaping (Result<T.ResponseType?, CoreError>, CorrelationId?) -> Void
+    ) {
```

Comment:
Nitpick - this spacing is strange to me.

I feel that one line is fine, since this isn't that long
```
    public func fetch<T: APIRequest>(endpoint: T, completion: @escaping (Result<T.ResponseType?, CoreError>, CorrelationId?) -> Void) {
```

If not, then two lines with `completion` starting the new line and lining up in the `:` for the input params.

Reply:
I could be wrong, but I think lining up `:` is more of an ObjC convention isn't it? 
But I'm down having this one line :smile:

Reply:
Wouldn't this hit our `line_length` linter warning as one line? The [default is 120 characters](https://realm.github.io/SwiftLint/line_length.html) and this looks to be over that. I think maybe this example would be a good reason run our linter during build in Xcode since there are things like this that we may only potentially catch once pushing to GitHub. But that could be a separate discussion. 😄

Reply:
> I could be wrong, but I think lining up : is more of an ObjC convention isn't it?

I thought it was application to Swift as well. I think having the `(` and `)` on different lines is strange and hard to read.

Reply:
lining up the `:` is definitely more of an obj-c thing - xcode doesn't align newlined parameters like this by default, and I haven't seen this before in swift

re: newlining parameters - This was new to me when I came to paypal, but I've grown to really appreciate the way it visually organizes the code. for reference, this is also part of the google [swift style guide](https://google.github.io/swift/#line-wrapping). either every parameter gets wrapped to a new line, or everything should fit on a single line - this can also prevent blocks of code hanging in freespace in the middle of the page,  a la 

```swift
    func someFunctionWithAVeryLongName(and many: String,
                                       additional: String,
                                       parameters: String) {
    }
```

Reply:
Gotcha - thanks for the resources/ comments. Makes sense to me!